### PR TITLE
Make call control buttons accessible to screen reader users

### PR DESCRIPTION
**Pull request by pvagner**
Perhaps it has just been overlooked.
Signed-off-by: Peter Vágner <pvdeejay@gmail.com>

Fixes https://github.com/vector-im/element-web/issues/18358

### CODE REVIEW

**Comment by t3chguy** [comment]
Can you run `yarn i18n`	and commit the changes please

**Comment by t3chguy** [comment]
I'm wondering whether these should change into Tooltip buttons so that it gains the Aria label & also a tooltip for sighted users, iconography alone is not an ideal situation

**Comment by pvagner** [comment]
@t3chguy  Hopefully addressed both points correctly.
Thank you

**Review by t3chguy — APPROVED** [review]
Code wise this is great, thanks, will run it by design on Monday as they may want to tweak copy

File: src/components/views/voip/CallView.tsx [hunk 1]
```diff
@@ -25,6 +25,8 @@ import RoomAvatar from "../avatars/RoomAvatar";
 import { CallState, CallType, MatrixCall, CallEvent } from 'matrix-js-sdk/src/webrtc/call';
 import classNames from 'classnames';
 import AccessibleButton from '../elements/AccessibleButton';
+import AccessibleTooltipButton from "../elements/AccessibleTooltipButton";
+import { ContextMenuTooltipButton } from "../../../accessibility/context_menu/ContextMenuTooltipButton";
 import {isOnlyCtrlOrCmdKeyEvent, Key} from '../../../Keyboard';
 import {alwaysAboveLeftOf, alwaysAboveRightOf, ChevronFace, ContextMenuButton} from '../../structures/ContextMenu';
```
Comment (t3chguy) [inline src/components/views/voip/CallView.tsx:31]:
```suggestion
import {isOnlyCtrlOrCmdKeyEvent, Key} from '../../../Keyboard';
import {
    alwaysAboveLeftOf,
    alwaysAboveRightOf,
    ChevronFace,
    ContextMenuTooltipButton,
} from '../../structures/ContextMenu';
```

Comment (t3chguy) [reply src/components/views/voip/CallView.tsx:31]: It should be re-exported by ContextMenu.

https://github.com/matrix-org/matrix-react-sdk/blob/f156c2db15e7e1bef5fd5e444be5dec644aac18d/src/components/structures/ContextMenu.tsx#L506

In any case, this suggestion is to fix the lint caused by an unused import (ContextMenuButton)

**Comment by t3chguy** [comment]
@pvagner any chance you can pull latest develop? Layering is not in our favour

**Comment by nadonomy** [comment]
@gaelledel are you ok to review this pls on behalf of @matrix-org/design while context switched into VoIP?

File: src/components/views/voip/CallView.tsx [hunk 2]
```diff
@@ -25,8 +25,14 @@ import RoomAvatar from "../avatars/RoomAvatar";
 import { CallState, CallType, MatrixCall, CallEvent } from 'matrix-js-sdk/src/webrtc/call';
 import classNames from 'classnames';
 import AccessibleButton from '../elements/AccessibleButton';
-import { isOnlyCtrlOrCmdKeyEvent, Key } from '../../../Keyboard';
-import { alwaysAboveLeftOf, alwaysAboveRightOf, ChevronFace, ContextMenuButton } from '../../structures/ContextMenu';
+import AccessibleTooltipButton from "../elements/AccessibleTooltipButton";
+import { isOnlyCtrlOrCmdKeyEvent, Key} from '../../../Keyboard';
```
Comment (SimonBrandner) [inline src/components/views/voip/CallView.tsx:29]:
```suggestion
import { isOnlyCtrlOrCmdKeyEvent, Key } from '../../../Keyboard';
```

**Comment by t3chguy** [comment]
@gaelledel any chance we could get review here before the pull request bit-rots?

Thanks

**Review by gaelledel — CHANGES_REQUESTED** [review]
We need to use the Element styling for tooltips and make sure that they appear above the button

**Comment by t3chguy** [comment]
> We need to use the Element styling for tooltips

This PR makes use of our standard AccessibleTooltipButton - so does use the same styling as the vast majority of our tooltips.

**Comment by gaelledel** [comment]
image (4)
Screenshot 2021-08-04 at 12 41 04

The styling of the tooltip should match the current web client styling on each UI theme.

**Comment by t3chguy** [comment]
@gaelledel it is exactly the same tooltip component as the `Dialpad` one you demonstrate. You're comparing Dark theme & Light theme though.

image
image

**Review by gaelledel — APPROVED** [review]

**Comment by t3chguy** [comment]
Have fixed merge conflicts, removed unused variable, fixed 2 inaccessible clickable divs and updated tooltip alignment to be closer to the design ask

image

File: src/components/views/voip/CallView.tsx [hunk 3]
```diff
@@ -479,9 +485,12 @@ export default class CallView extends React.Component<IProps, IState> {
         let vidMuteButton;
         if (this.props.call.type === CallType.Video) {
             vidMuteButton = (
-                <AccessibleButton
+                <AccessibleTooltipButton
                     className={vidClasses}
                     onClick={this.onVidMuteClick}
+                    title={this.state.vidMuted ? _t("Start the camera") : _t("Stop the camera")}
+                    alignment={Alignment.Top}
+                    yOffset={-24}
```
Comment (SimonBrandner) [inline src/components/views/voip/CallView.tsx:493]: This number seems to be always the same, could it perhaps be a const at the top of the file?

Comment (t3chguy) [reply src/components/views/voip/CallView.tsx:493]: yeah, good point

**Review by SimonBrandner — APPROVED** [review]

**Review by turt2live — APPROVED** [review]

**Pull request merged**